evalml/pipelines/components/transformers/transformer.py

```diff
@@ -73,3 +75,11 @@ def fit_transform(self, X, y=None):
 
     def _get_feature_provenance(self):
         return {}
+
+
+class TargetTransformer(Transformer):
+    """A component that transforms the target."""
+
+    @abstractmethod
+    def inverse_transform(self, y):
```

Comment:
Could be useful in docstring here to specify arguments + what we expect to return for when people override :)